library/core/src/result.rs

```diff
@@ -1281,16 +1281,26 @@ impl<T, E> Result<T, E> {
     ///
     /// # Examples
     ///
-    /// Basic usage:
+    /// ```
+    /// fn squared_string(x: u32) -> Result<String, &'static str> {
+    ///     Ok((x * x).to_string())
```

Comment:
suggestion: Maybe tweak this a bit to be something that can show this failing too?

Like if it were
```rust
    let sq = x.checked_mul(x).ok_or("overflowed")?;
    Ok(sq)
```
then it could have another example of
```
assert_eq!(Ok(1_000_000).and_then(squared_string), Err("not a number"));
```

Reply:
Ah nice. Good improvement.

Reply:
Oh and I made a similar edit to `Option::and_then`, in pretty much the same spirit.